h/stcbackground.h

```diff
@@ -0,0 +1,54 @@
+/*
+  This program and the accompanying materials are
+  made available under the terms of the Eclipse Public License v2.0 which 
+  accompanies this distribution, and is available at 
+  https://www.eclipse.org/legal/epl-v20.html
+
+  SPDX-License-Identifier: EPL-2.0
+
+  Copyright Contributors to the Zowe Project.
+*/
+
+#ifndef __STCBACKGROUND__
+#define __STCBACKGROUND__
+
+#include "stcbase.h"
+#define STC_BG_CB_LABEL_LEN 30
+
+struct STCIntervalCallbackData_tag;
+typedef int (*STCIntervalCallback)(void* server, 
+                                    STCBase* stcBase, 
+                                    STCModule* module, 
+                                    struct STCIntervalCallbackData_tag* callbackData, 
+                                    void* userData
+                                   );
+
+typedef struct STCIntervalCallbackData_tag {
+  int id;
+  char callbackLabel[STC_BG_CB_LABEL_LEN];
+  STCIntervalCallback callback;
+  void* userData;
+  int timeInterval;
+  int countInterval;
+} STCIntervalCallbackData; 
+
+typedef struct STCCallbackList_tag {
+  void* server;
+  void* callbackList;
+} STCCallbackList;
+
+STCModule* stcInitBackgroundModule(void* server, STCBase *base);
+int stcAddIntervalCallback(STCModule *module, STCIntervalCallback callback, char* callbackLabel, int timeInterval, void* userData);
+int stcModifyInterval(STCIntervalCallbackData* callback, int newInterval);
```

Comment:
Let's not add public functions if we don't need them. 

This function exposes `STCIntervalCallbackData` which shouldn't be exposed. You've already identified your callback using a label, so we'd want to keep using it to address the callback.

Reply:
i see, good choice

just one question though:
what about exposing it here
```
typedef int (*STCIntervalCallback)(void* server,
                                  struct STCBase_tag *stcBase,
                                  struct STCModule_tag *module, 
                                  STCIntervalCallbackData* callbackData, 
                                  void *userData);
```

Reply:
I think we should but we shouldn't expose the field in the header. That way we can change the internal implementation and provide public functions to access them. This will ensure backward/forward compatibility.

Reply:
hi @ifakhrutdinov , i believe then we should remove it from signature as well, as public functions can use callback label, so reference is not needed.

Moved all structure definitions to .c

Reply:
its in alignment with your comment above
>You've already identified your callback using a label, so we'd want to keep using it to address the callback.

Reply:
You can keep it in the signature of `STCIntervalCallback` by using forward declaration. Concrete callbacks can use it to access internal stuff via public functions. Concrete callbacks don't usually know their label. The label is useful for functions like "removeCallback".